File: BaseLibrary/communication/command/common/MerchantCommand.cs
using System;
using System.Collections.Generic;

namespace BaseLibrary.communication.command.common {
    /// <summary>
    /// Command for specific what robot want to buy.
    /// </summary>
    public class MerchantCommand : ACommonCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        private int _motorId;
        /// <summary>
        /// What motor want to robot buy.
        /// </summary>
        public int MOTOR_ID {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _motorId;
            }
            private set => _motorId = value;
        }

        private int _classEquipmentId;
        /// <summary>
        /// What class equipment want to robot buy.
        /// </summary>
        public int CLASS_EQUIPMENT_ID {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _classEquipmentId;
            }
            private set => _classEquipmentId = value;
        }

        private int _armorID;
        /// <summary>
        /// What armor want to robot buy.
        /// </summary>
        public int ARMOR_ID {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _armorID;
            }
            private set => _armorID = value;
        }

        /// <summary>
        /// How many HP want robot to fix.
        /// </summary>
        public int REPAIR_HP { get; private set; }

        public MerchantCommand(int motorId, int armorId, int classEquipmentId, int repairHp) {
            MOTOR_ID = motorId;
            CLASS_EQUIPMENT_ID = classEquipmentId;
            ARMOR_ID = armorId;
            REPAIR_HP = repairHp;
            pending = false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/common/MerchantCommand.java

package BaseLibrary.communication.command.common;

import BaseLibrary.communication.protocol.AProtocol;

/**
 * Command for specific what robot want to buy.
 */
public class MerchantCommand extends ACommonCommand {

	private static final String NAME = "MERCHANT";

	static {
		AProtocol.registerForDeserialize(NAME, MerchantCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}

	private int motorID;

	/**
	 * What motor want to robot buy.
	 */
	public int getMotorID() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return motorID;
	}

	private int classEquipmentID;

	/**
	 * What class equipment want to robot buy.
	 */
	public int getClassEquipmentID() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return classEquipmentID;
	}

	private int armorID;

	/**
	 * What armor want to robot buy.
	 */
	public int getArmorID() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return armorID;
	}

	private int repairHP;

	/**
	 * How many HP want robot to fix.
	 */
	public int getRepairHP() {
		return repairHP;
	}

	public MerchantCommand(int motorID, int armorID, int classEquipmentID, int repairHP) {
		this.motorID = motorID;
		this.classEquipmentID = classEquipmentID;
		this.armorID = armorID;
		this.repairHP = repairHP;
		pending = false;
	}
}
